### ISSUE TITLE
Can not set `keyGeneratorDelimiter` to empty string

### ISSUE THREAD

**Issue by xavierchow** [couchbaselabs/node-ottoman#618]
I'd like to set the key of the document to the `id` without the prefix, so I set `keyGenerator` to return emtpy string`''` and also tried to set `keyGeneratorDelimiter` to empty string `''` .

However, I found the key of the created document in the bucket is still with the default value `KEY_GENERATOR_DELIMITER`. i.e. `::`.
![image](https://user-images.githubusercontent.com/2748884/147875684-42aaf4ca-9690-4c64-ad8b-f47116012139.png)

Looking into the code, it seems passing empty string to `delimeter` doesn't work, because `''` is falsy in JavaScript so it always takes the default value `KEY_GENERATOR_DELIMITER` as the parameter, is it expected? 
```javascript
export const _keyGenerator = (keyGen, { metadata, id }, delimeter = KEY_GENERATOR_DELIMITER) => {
  const prefix = keyGen({ metadata });
  const prefixDelimeter = `${prefix || ''}${delimeter || ''}`;
  if (prefixDelimeter && id.toString().startsWith(prefixDelimeter)) {
    return id;
  }
  return `${prefixDelimeter}${id}`;
};
```
https://github.com/couchbaselabs/node-ottoman/blob/91a35ed2182cb3b5163613e627097c90ab2020fe/src/utils/constants.ts?_pjax=%23repo-content-pjax-container#L42-L48

I think rewriting the `_keyGenerator` function as follows would solve my problem though, any ideas?

```
export const _keyGenerator = (keyGen, { metadata, id }, delimeter) => {
  const prefix = keyGen({ metadata });
  const prefixDelimeter = `${prefix || ''}${delimeter == null ? KEY_GENERATOR_DELIMITER  || delimeter}`;
  if (prefixDelimeter && id.toString().startsWith(prefixDelimeter)) {
    return id;
  }
  return `${prefixDelimeter}${id}`;
};
```

**Comment by ejscribner** [comment]
Hey @xavierchow, thanks so much for raising this issue! 

I believe you are correct; there should be an option to remove the prefix altogether (set key=id), but it currently defaults to prepending `::` even if an empty string is passed in. 

I'll look into adding the suggested changes and hopefully we can get this taken care of in a release sometime later this week or early next week.

Thanks again for being a member of our growing community!

**Comment by ejscribner** [comment]
Hey @xavierchow. Wanted to update you that this functionality has been added in the latest release of Ottoman v2.1.0, available now on NPM. Thanks!

**Comment by xavierchow** [comment]
fixed with v2.1.0, closing.
